EpikCord/client/http_client.py

```diff
@@ -1,312 +1,233 @@
-
-if TYPE_CHECKING:
-    import discord_typings
-
-
-class _FakeTask:
-    def cancel(self):
-        return True
-
-
-class UnknownBucket:
+class MockBucket:
     def __init__(self):
-        self.event = asyncio.Event()
-        self.event.set()
-        self.close_task: Union[_FakeTask, asyncio.Task] = _FakeTask()
-
+        self.internal_hash: str = ""
+        self.external_hash: str = ""
 
-class Bucket(UnknownBucket):
-    def __init__(self, *, discord_hash: str):
-        super().__init__()
-        self.bucket_hash = discord_hash
+    async def wait(self):
+        ...
 
-    def __eq__(self, other):
-        return self.bucket_hash == other.bucket_hash
+    def clear(self):
+        ...
 
-    def __hash__(self):
-        return id(self)
+    def set(self):
+        ...
 
+    def __eq__(self, other: MockBucket):
+        if isinstance(other, MockBucket):
+            return True
+        return False
 
-class DiscordWSMessage:
-    def __init__(self, *, data, type, extra):
-        self.data = data
-        self.type = type
-        self.extra = extra
 
-    def json(self) -> Any:
-        return json.loads(self.data)
-
-
-class GatewayWebsocket(ClientWebSocketResponse):
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.buffer: bytearray = bytearray()
-        self.inflator = zlib.decompressobj()
+class Bucket:
+    def __init__(self, *, bucket_hash: str):
+        """
+        Parameters
+        ----------
+        bucket_hash: str
+            The bucket hash of the bucket provided by Discord.
+
+        Attributes
+        ----------
+        bucket_hash: str
+            The bucket hash of the bucket provided by Discord.
+        event: asyncio.Event
+            The event that is used to wait for the bucket to be set.
+        """
+        self.hash: str = bucket_hash
+        self.event: asyncio.Event = asyncio.Event()
+        self.event.set()
 
-    async def receive(self, *args, **kwargs):
-        ws_message = await super().receive(*args, **kwargs)
-        message = ws_message.data
+    async def wait(self):
+        """
+        Waits for the bucket to be set.
+        """
+        await self.event.wait()
 
-        if isinstance(message, bytes):
+    def set(self):
+        """
+        Sets the bucket.
+        """
+        self.event.set()
 
-            self.buffer.extend(message)
+    def clear(self):
+        """
+        Clears the bucket.
+        """
+        self.event.clear()
 
-            if len(message) < 4 or message[-4:] != b"\x00\x00\xff\xff":
-                return
+    def __eq__(self, other: Bucket):
+        if isinstance(other, Bucket):
+            return self.hash == other.hash
+        return False
 
-            message = self.inflator.decompress(self.buffer)
 
-            message = message.decode("utf-8")
-            self.buffer: bytearray = bytearray()
+class TopLevelBucket(Bucket):
+    def __init__(self, *, major_parameters: Dict[str, Any]):
```

Comment:
Why not accept each major parameter as keyword arguments? It'll be much cleaner.

Reply:
As I was writing my reply, I realised that I don't even need the major parameters.

Reply:
It's very messy to follow through with.